src/io/flutter/project/FlutterIconProvider.java

```diff
@@ -11,30 +11,45 @@
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.psi.PsiDirectory;
 import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiInvalidElementAccessException;
 import com.jetbrains.lang.dart.DartFileType;
 import com.jetbrains.lang.dart.psi.DartFile;
+import icons.FlutterIcons;
 import io.flutter.FlutterUtils;
 import io.flutter.pub.PubRoot;
 import io.flutter.utils.FlutterModuleUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import javax.swing.*;
+import java.util.Objects;
 
 import static com.intellij.psi.impl.ElementBase.overlayIcons;
 
 public class FlutterIconProvider extends IconProvider {
 
   private static final Icon TEST_FILE = overlayIcons(DartFileType.INSTANCE.getIcon(), AllIcons.Nodes.JunitTestMark);
 
+
   @Nullable
   public Icon getIcon(@NotNull final PsiElement element, @Iconable.IconFlags final int flags) {
     final boolean hasFlutterModule = FlutterModuleUtils.hasFlutterModule(element.getProject());
+    if (!hasFlutterModule) return null;
+
     // Directories.
-    if (hasFlutterModule && element instanceof PsiDirectory) {
+    if (element instanceof PsiDirectory) {
       final VirtualFile file = ((PsiDirectory)element).getVirtualFile();
       if (!file.isInLocalFileSystem()) return null;
 
+      // Project root.
+      try {
+        if (Objects.equals(file.getPath(), element.getProject().getBaseDir().getPath())) {
```

Comment:
I feel like there's got to be a better way to compare virtual files...  Looking.